src/Sweep.cpp

```diff
@@ -290,666 +290,590 @@ void InfectSweep(double t, int run) // added run number as argument in order to
-						} while (f);
+							else									KeepSearching = 0;
+						} while (KeepSearching);
 					}
-					f = (StateT[tn].cell_inf[j] < 0); //// flag for whether infector j had their place(s) closed. <0 (true) = place closed / >=0 (false) = place not closed. Set in if (sbeta > 0) part of loop over infectious people.
-					// ci is the index of the jth infectious person in the cell
-					ci = c->infected[j];
-					// si is the jth selected person in the cell
-					Person* si = Hosts + ci;
+					int InfectorPlaceClosedFlag = (StateT[ThreadNum].cell_inf[PotentialInfector_CellIndex] < 0); //// flag for whether infector PotentialInfector_CellIndex had their place(s) closed. <0 (true) = place closed / >=0 (false) = place not closed. Set in if (SpatialSeasonal_Beta > 0) part of loop over infectious people.
 
-					//calculate flag (fct) for digital contact tracing here at the beginning for each individual infector
-					int fct = ((P.DoDigitalContactTracing) && (t >= AdUnits[Mcells[si->mcell].adunit].DigitalContactTracingTimeStart)
-						&& (t < AdUnits[Mcells[si->mcell].adunit].DigitalContactTracingTimeStart + P.DigitalContactTracingPolicyDuration) && (Hosts[ci].digitalContactTracingUser == 1)); // && (ts <= (Hosts[ci].detected_time + P.usCaseIsolationDelay)));
+					int PotentialInfector_Index = ThisCell->infected[PotentialInfector_CellIndex];
+					// PotentialInfector_Spatial is the jth infected person in the cell
+					Person* PotentialInfector_Spatial = Hosts + PotentialInfector_Index;
 
+					//calculate flag (DigiContactTrace_ThisPersonNow) for digital contact tracing here at the beginning for each individual infector
+					int DigiContactTrace_ThisPersonNow = ((P.DoDigitalContactTracing) && (t >= AdUnits[Mcells[PotentialInfector_Spatial->mcell].adunit].DigitalContactTracingTimeStart)
+						&& (t < AdUnits[Mcells[PotentialInfector_Spatial->mcell].adunit].DigitalContactTracingTimeStart + P.DigitalContactTracingPolicyDuration) && (Hosts[PotentialInfector_Index].digitalContactTracingUser == 1)); // && (TimeStepNow <= (Hosts[PotentialInfector_Spatial].detected_time + P.usCaseIsolationDelay)));
 
 					//// decide on infectee
 					
-					// do the following while f2 = 1 
+					int KeepSearchingForCellToInfect = 1;  // do the following while KeepSearchingForCellToInfect = 1
 					do
 					{
 						//// chooses which cell person will infect
-						// pick random s between 0 and 1
-						s = ranf_mt(tn);
+						// pick RandomNum between 0 and 1
+						double RandomNum = ranf_mt(ThreadNum);
 						// generate l using InvCDF of selected cell and random integer between 0 and 1024
-						int l = c->InvCDF[(int)floor(s * 1024)];
-						// loop over c->cum_trans array until find a value >= random number s
-						while (c->cum_trans[l] < s) l++;
+						int l = ThisCell->InvCDF[(int)floor(RandomNum * 1024)];
+						// loop over ThisCell->cum_trans array until find a value >= RandomNum
+						while (ThisCell->cum_trans[l] < RandomNum) l++;
```

Comment:
Might as well zap this "l" at some point.
The two flags (KeepSearching...) and DigiContactTrace_ThisPersonNow could be bools